### DISCUSSION TITLE
Make pull-payment request disabled when archived

### DISCUSSION THREAD

**Discussion by mp21-org** [btcpayserver/btcpayserver#3691] [Ideas / Feature Requests]
It seems to be still possible to claim pull-payment requests (if I have the URL), even if they where archived.

Together with the auto aproved payment processor, this could lead to problematic payments. Is there a plan to offer deleting/disabeling/invalidating pull-payment requests?

**Comment by bolatovumar** [comment]
@pavlenex any thoughts on this issue? Seems like it could become a problem if someone were to have auto-approval enabled on a pull payment enabled and archive it thinking that it wouldn't be possible to claim but that's not true and those who have the ULR can still claim it anyway.

**Comment by pavlenex** [comment]
Hm, I am not quite sure, archiving isn't deleting a pull request, archive in a way just cleans a pull payment out of the view.

@NicolasDorier Currently, when you create a pull payment, and archive it, someone with a link can still claim it, is this something expected or a bug? We use the same logic for the invoice, if an invoice is archived, it can still be paid, but being paid and getting money out of your wallet aren't really the same.

**Reply by bolatovumar** [reply]
Yeah, it's not really clear to me what the expectation for "archiving" is. It may sound to some people as if that completely disables the pull payment but that's not what it does at all.

**Reply by dstrukt** [reply]
Agreed .. even if it's not what was originally intended, as a user I would expect "archiving" the pull payment would disable the ability to claim funds.

If "Archive" proves to be an inaccurate label, it would make sense to rename with a singular CTA that encompassed both the "Disable / Archive" functionality.
